core/modules/sql/SqlConnection.cc

```diff
@@ -126,14 +127,29 @@ SqlConnection::~SqlConnection() {
 
 bool
 SqlConnection::connectToDb(SqlErrorObject& errObj) {
-    assert(_connection.get());
-    if (_connection->connected()) return true;
+    assert(_connection != nullptr);
+    static int lastChecked = 0;
+    time_t currentTime = time(0);
+
+    if (_connection->connected()) {
+        int rc = mysql_ping(_connection->getMySql());
+        if (rc == 0) {
+            if (currentTime < lastChecked+60) return true;
```

Comment:
The intention was not to limit ping-check rate but to make sure the connection actually existed, as `_connection->connected()` just reads off the value of a saved class variable (bool) initiated when the connection was first made. A "timeout" would've made it so that variable would still say `true` even after the connection had timed out. 

As it is applied here I agree it doesn't do anything since I convoluted the multiple commits from my original test branch. I'll make sure it's fine to just get rid of the timing stuff entirely and clean up.